src/main/java/com/openlattice/shuttle/dates/DecadeChangeHelper.java

```diff
@@ -0,0 +1,56 @@
+package com.openlattice.shuttle.dates;
+
+import com.openlattice.shuttle.util.Cached;
+import com.openlattice.shuttle.util.Constants;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.OffsetDateTime;
+
+public class DecadeChangeHelper {
+    /**
+     *
+     * Dealing with the fact that a two-year year-pattern should not all be in 20YY
+     *
+     */
+    public static boolean checkDatePatternIsTwoDigitYear( String datePattern ) {
+        boolean yyMatch = Cached.getMatcherForString( datePattern, ".*yy.*" ).matches();
+        boolean yyyyMatch = Cached.getMatcherForString( datePattern, ".*yyyy.*" ).matches();
+        return yyMatch && !yyyyMatch;
```

Comment:
would be more performant to only test one regex, like
```
return Cached.getMatcherForString( datePattern, ".*([^y]|^)yy([^y]|$).*" ).matches();
```
or (longer but more readable regex):
```
boolean yearLengthTester = Cached.getMatcherForString( datePattern, ".*(yy)(y+)?.*"
return yearLengthTester.matches() && yearLengthTester.group( 2 ) == null;
```